cloud-sql/mysql/Startup.cs

```diff
@@ -55,152 +54,157 @@ public void ConfigureServices(IServiceCollection services)
                     options.Version = "Test";
                 });
             }
-            services.AddScoped(typeof(DbConnection), (IServiceProvider) =>
-                InitializeDatabase());
+            services.AddScoped(typeof(MySqlConnectionStringBuilder),
+                (IServiceProvider) => new MySqlConnect().GetMySqlConnectionString());
+            services.AddScoped<MySqlConnect>();
             services.AddMvc(options =>
             {
                 options.Filters.Add(typeof(DbExceptionFilterAttribute));
             });
             _services = services;
         }
 
-        DbConnection InitializeDatabase()
+        public class MySqlConnect
         {
-            DbConnection connection;
+            public DbConnection GetMySqlConnection(
+                MySqlConnectionStringBuilder connectionString)
+            {
+                // [START cloud_sql_mysql_dotnet_ado_backoff]
+                var connection = Policy
+                    .HandleResult<DbConnection>(conn => conn.State != ConnectionState.Open)
+                    .WaitAndRetry(new[]
+                    {
+                    TimeSpan.FromSeconds(1),
+                    TimeSpan.FromSeconds(2),
+                    TimeSpan.FromSeconds(5)
+                    }, (result, timeSpan, retryCount, context) =>
+                    {
+                        // Log any warnings here.
+                    })
+                    .Execute(() =>
+                    {
+                        // Return a new connection.
+                        return new MySqlConnection(connectionString.ConnectionString);
+                    });

```

Comment:
I'm pretty sure this code is not adding anything (it wasn't adding anything before either). The connection returned in the `Execute` portion of the policy is always closed. So the retry will always happen for the three intervals specified there and will end up returning a closed connection anyway, which is fine, because the connection is being opened on the controller before usage.
If you meant to retry attempting to open the connection, then maybe the policy needs to actually attempt to open the connection on the execute portion and handle the result when there's a [MySqlException](https://dev.mysql.com/doc/dev/connector-net/8.0/html/M_MySql_Data_MySqlClient_MySqlConnection_Open.htm) thrown. It'll have to be slightly different for each of the three providers, but the spirit is the same.

(You can add a breakpoint on line 87 and see how it always executes 3 times whenever you attempt to get a connection).

Reply:
Done.